File: solutions/709_to_lower_case.cpp
#include "aux.cpp"
/**
Implement function ToLowerCase() that has a string parameter str, and returns
the same string in lowercase.

Example 1:
Input: "Hello"
Output: "hello"
Example 2:
Input: "here"
Output: "here"
Example 3:
Input: "LOVELY"
Output: "lovely"

 https://leetcode.com/problems/to-lower-case/description/
 **/
static const int _ = []() {
  ios::sync_with_stdio(false);
  cin.tie(NULL);
  return 0;
}();

class Solution {
public:
  string toLowerCase(string str) {
    transform(str.begin(), str.end(), str.begin(), ::tolower);
    return str;
  }
};

int main() {
  Solution s;
  string str = "aHKKKldkjf";
  say(s.toLowerCase(str));
}

File: python_solutions/709.to-lower-case.rb
#
# @lc app=leetcode id=709 lang=ruby
#
# [709] To Lower Case
#
# https://leetcode.com/problems/to-lower-case/description/
#
# algorithms
# Easy (74.99%)
# Total Accepted:    52.8K
# Total Submissions: 70.4K
# Testcase Example:  '"Hello"'
#
# Implement function ToLowerCase() that has a string parameter str, and returns
# the same string in lowercase.
# 
# 
# 
# 
# Example 1:
# 
# 
# Input: "Hello"
# Output: "hello"
# 
# 
# 
# Example 2:
# 
# 
# Input: "here"
# Output: "here"
# 
# 
# 
# Example 3:
# 
# 
# Input: "LOVELY"
# Output: "lovely"
# 
# 
# 
# 
# 
#
# @param {String} str
# @return {String}
def to_lower_case(str)
    str.downcase
end

str = 'LOVELY'
p to_lower_case(str)
